joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/passes/ast/AstCreationPassTests.scala

```diff
@@ -1953,6 +1950,40 @@ class AstCreationPassTests extends AbstractPassTest {
       macFixture.close()
       linuxFixture.close()
     }
+
   }
 
+  "AST with types" should {
+
+    "be correct for function edge case" in AstFixture("class Foo { char (*(*x())[5])() }", "test.cpp") { cpg =>
+      val List(method) = cpg.method.nameNot("<global>").l
+      method.name shouldBe "x"
+      method.fullName shouldBe "Foo.x"
+      method.code shouldBe "char (*(*x())[5])()"
+      method.signature shouldBe "char Foo.x ()"
+    }
+
+    "be consistent with pointer types" in AstFixture("""
+        |struct x { char * z; };
+        |char *a(char *y) {
+        |  char *x;
+        |}
+      """.stripMargin) { cpg =>
+      cpg.member.name("z").typeFullName.head shouldBe "char"
+      cpg.parameter.name("y").typeFullName.head shouldBe "char"
+      cpg.local.name("x").typeFullName.head shouldBe "char"
+      cpg.method.name("a").methodReturn.typeFullName.head shouldBe "char"
```

Comment:
We do not append the pointer * to the type fullname.
1) we do not have a suitable representation for pointer types (type decls)
2) we won't find the corresponding typedecl at the backend without custom/adapted logic which would be C/C++ specific
3) does not have any impact on DF analysis
4) you can always have a look into the code field (e.g., locals) to identify if its a pointer (for AST-based queries)

Reply:
I would much prefer having the `*` here. I want to do things like find all parameters that are pointers. Until we have a proper way to represent structural types in the CPG, checking whether the type name ends with a `*` would be a decent approximation for that. Parsing the code field would be much more annoying because then I'd have to figure out how to strip the non-type information from that code field (the name is usually in the end, except when it isn't...).

In the backend I think without any work we'd just end up creating stub typedecls for all these pointer types. Those won't have the information that `char*` is in any way related to `char`. While that would be nice to have, we're already in the exact same situation for e.g. `char[1]` or `signed char` and I'm not aware of it being much of a problem.

Reply:
```
class Foo { ... }

Foo *a() {
  return ...;
}
```
would result in a return type `Foo*` which we simply do not have. We have only `Foo`.

Reply:
How is this different for `Foo[1]`? Won't the `TypeDeclStubCreator` just create it on-demand?

Reply:
`Foo[1]` is actually a different type (an array of `Foo`).
Yes, the `TypeDeclStubCreator` would create it but we have no link between `Foo*` and `Foo` then for tracking.
Btw: we already had this discussion here in the Joern issues a few times (see Fabs comments).

Reply:
Remaining pointer inconsistencies are resolved with the latest commit here.
Please have a look.